client/app/core/authentication-api.factory.js

```diff
@@ -10,13 +10,51 @@ function base64encode (str, encoding = 'utf-8') {
 /** @ngInject */
 export function AuthenticationApiFactory ($http, API_BASE, Session, Notifications) {
   var service = {
+    globalLogin: globalLogin,
     login: login,
+    oidcLogin: oidcLogin
   };
 
   return service;
 
+  function globalLogin (authMode, userLogin, password, access_token) {
+    if (authMode == 'oidc') {
+      self.currentAuthMode = authMode
+      return oidcLogin(access_token)
+    } else {
+      self.currentAuthMode = 'database'
+      return login(userLogin, password)
+    }
+  }
+
+  function oidcLogin (access_token) {
+    return new Promise((resolve, reject) => {
```

Comment:
this is not needed, `loginSuccess` can just `return response` to achieve the same result by just `return $http.get...`

Reply:
👍   much simpler. thank you !!